lib/flight_scheduler/schedulers/fifo_scheduler.rb

```diff
@@ -28,90 +28,12 @@
 class FifoScheduler
   FlightScheduler.app.schedulers.register(:fifo, self)
 
-  def initialize
-    @group_id_queue = Concurrent::Array.new([])
-    @data = Concurrent::Map.new
-    @allocation_mutex = Mutex.new
-  end
-
-  # NOTE: This method is only intended for public consumption about the
-  #       current state of the running jobs
-  #
-  #       It should not be used internally as it is a composite of the
-  #       internal state
-  def queue
-    # TODO: Consider switching to a shared-locking semaphore here. It does not
-    #       matter if other threads are reading at the same time, however it
-    #       does need to block the allocation of new jobs
-    @allocation_mutex.synchronize do
-      @group_id_queue.map do |id|
-        active  = @data[id][:active]
-        job     = @data[id][:job]
-
-        if active.empty?
-          job
-        elsif @data[id][:enum].peek
-          [job, active]
-        else
-          active
-        end
-      end.flatten
-    end
-  end
+  class InvalidJobType < RuntimeError; end
```

Comment:
:panda_face: We should probably look at condensing all the errors into a single inheritance hierarchy at some point, but not a huge priority right now

Reply:
There's a lot to be said for namespacing exceptions inside the classes that throw them.  For instance `JobRegistry::DuplicateJob` reads nicely and better than any non-namespaced name I can think of does.

Having them all inherit from a common base class could be done without changing any namespacing.

But not one for now.